internal/services/inference/deployment.go

```diff
@@ -0,0 +1,349 @@
+		publicEndpointMap := publicEndpoint.([]interface{})[0].(map[string]interface{})
+		if publicEndpointMap["is_enabled"].(bool) {
+			publicEp := inference.EndpointSpec{
+				Public:      &inference.EndpointSpecPublic{},
+				DisableAuth: publicEndpointMap["disable_auth"].(bool),
+			}
+			endpoints = append(endpoints, &publicEp)
+		}
+	}
+
+	if privateEndpoint, ok := d.GetOk("private_endpoint"); ok {
+		privateEndpointMap := privateEndpoint.([]interface{})[0].(map[string]interface{})
+		if privateID, exists := privateEndpointMap["private_endpoint_id"]; exists {
+			privateEp := inference.EndpointSpec{
+				PrivateNetwork: &inference.EndpointSpecPrivateNetwork{
+					PrivateNetworkID: regional.ExpandID(privateID.(string)).ID,
+				},
+				DisableAuth: privateEndpointMap["disable_auth"].(bool),
+			}
+			endpoints = append(endpoints, &privateEp)
+		}
+	}
+
+	return endpoints
+}
+
+func ResourceDeploymentRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api, region, id, err := NewAPIWithRegionAndID(m, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	deployment, err := waitForDeployment(ctx, api, region, id, d.Timeout(schema.TimeoutRead))
+	if err != nil {
+		if httperrors.Is404(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("name", deployment.Name)
+	_ = d.Set("region", deployment.Region)
+	_ = d.Set("project_id", deployment.ProjectID)
+	_ = d.Set("node_type", deployment.NodeType)
+	_ = d.Set("model_name", deployment.ModelName)
+	_ = d.Set("min_size", int(deployment.MinSize))
+	_ = d.Set("max_size", int(deployment.MaxSize))
+	_ = d.Set("size", int(deployment.Size))
+	_ = d.Set("status", deployment.Status)
+	_ = d.Set("model_id", deployment.ModelID)
+	_ = d.Set("created_at", types.FlattenTime(deployment.CreatedAt))
+	_ = d.Set("updated_at", types.FlattenTime(deployment.UpdatedAt))
+	var privateEndpoints []map[string]interface{}
+	var publicEndpoints []map[string]interface{}
+
+	for _, endpoint := range deployment.Endpoints {
+		if endpoint.PrivateNetwork != nil {
+			privateEndpointSpec := map[string]interface{}{
+				"id":                  endpoint.ID,
+				"private_endpoint_id": regional.NewID(deployment.Region, endpoint.PrivateNetwork.PrivateNetworkID).String(),
+				"disable_auth":        endpoint.DisableAuth,
+				"url":                 endpoint.URL,
+			}
+			privateEndpoints = append(privateEndpoints, privateEndpointSpec)
+		}
+		if endpoint.PublicAccess != nil {
+			publicEndpointSpec := map[string]interface{}{
+				"id":           endpoint.ID,
+				"is_enabled":   true,
+				"disable_auth": endpoint.DisableAuth,
+				"url":          endpoint.URL,
+			}
+			publicEndpoints = append(publicEndpoints, publicEndpointSpec)
+		}
+	}
+
+	if len(privateEndpoints) > 0 {
+		_ = d.Set("private_endpoint", privateEndpoints)
+	}
+	if len(publicEndpoints) > 0 {
+		_ = d.Set("public_endpoint", publicEndpoints)
+	}
```

Comment:
MaxItems is 1 for `private_endpoint` and `public_endpoint`. This code make it seems multiple endpoints are supported. Should we let user specify multiple endpoints in terraform ? If API allows it, terraform provider would be ready.

Reply:
You can have only one private network and only one public